### PR TITLE
board/rtl8721csm/src/component/common : RTL8721CSM WPA3 support.

### PR DESCRIPTION
**Pull request by fengyunlai**
1st commit for RTL8721CSM WPA3 support, adding WPA3 cypto, auth type handle.

Signed-off-by: fengyunlai <1063072208@qq.com>

### CODE REVIEW

File: os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c [hunk 1]
```diff
@@ -422,12 +426,11 @@ int8_t cmd_wifi_connect(trwifi_ap_config_s *ap_connect_config, void *arg)
 	wifi_utils_ap_auth_type_e auth = ap_connect_config->ap_auth_type;
 	wifi_utils_ap_crypto_type_e crypto = ap_connect_config->ap_crypto_type;
 	/* WIFI_MANAGER_CRYPTO_AES: 3 */
-	if (crypto == 3 && auth != WIFI_UTILS_AUTH_WPA2_PSK) {
+	if (crypto == 3 && ((auth != WIFI_UTILS_AUTH_WPA2_PSK) && (auth != WIFI_UTILS_AUTH_WPA3_PSK))) {
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c:429]: Is this correct to check `(auth != WIFI_UTILS_AUTH_WPA2_PSK) && (auth != WIFI_UTILS_AUTH_WPA3_PSK)`? not `(auth != WIFI_UTILS_AUTH_WPA2_PSK) || (auth != WIFI_UTILS_AUTH_WPA2_PSK)`?
I think `auth` cannot be `WIFI_UTILS_AUTH_WPA2_PSK` and `WIFI_UTILS_AUTH_WPA2_PSK` simultaneuosly.

Comment (fengyunlai) [reply os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c:429]: Sorry I don't get what you mean. Is there a type error in you comment? My logic is as below, when  cypto is 3, which is aes, auth should be WPA2_PSK or WPA3_PSK, if not, return error

Comment (jeongchanKim) [reply os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c:429]: Oh sorry. This is correct. I was confused. Let's keep this.

Comment (fengyunlai) [reply os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c:429]: No problem~

**Comment by sunghan-chang** [comment]
@fengyunlai Is this enough to support WPA2? only change in 2 files? no additional library?

**Comment by fengyunlai** [comment]
> @fengyunlai Is this enough to support WPA2? only change in 2 files? no additional library?

Yes Mr Chang, current lib already support WPA3, further more, our latest lib are going to provide in WIFI performance PR Keerthana provide.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**